src/univariate/continuous/loguniform.jl

```diff
@@ -0,0 +1,66 @@
+"""
+    LogUniform(a,b)
+
+A positive random variable `X` is log-uniformly with parameters `a` and `b` if the logarithm of `X` is `Uniform(log(a), log(b))`.
+The *log uniform* distribution is also known as *reciprocal distribution*.
+```julia
+LogUniform(1,10)
+```
+External links
+
+* [Log uniform distribution on Wikipedia](https://en.wikipedia.org/wiki/Reciprocal_distribution)
+"""
+struct LogUniform{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+    LogUniform{T}(a::T, b::T) where {T <: Real} = new{T}(a, b)
+end
+
+function LogUniform(a::T, b::T; check_args=true) where {T <: Real}
+    check_args && @check_args(LogUniform, 0 < a < b)
+    LogUniform{T}(a, b)
+end
+
+LogUniform(a::Real, b::Real) = LogUniform(promote(a, b)...)
+
+function convert(::Type{LogUniform{T}}, d::LogUniform) where {T}
+    LogUniform(T(d.a), T(d.b))::LogUniform{T}
+end
+Base.minimum(d::LogUniform) = d.a
+Base.maximum(d::LogUniform) = d.b
+
+#### Parameters
+params(d::LogUniform) = (d.a, d.b)
+partype(::LogUniform{T}) where {T<:Real} = T
+
+#### Statistics
+
+function mean(d::LogUniform)
+    a, b = params(d)
+    (b - a) / log(b/a)
+end
+function var(d::LogUniform)
+    a, b = params(d)
+    log_ba = log(b/a)
+    (b^2 - a^2) / (2*log_ba) - ((b-a)/ log_ba)^2
+end
+mode(d::LogUniform)   = d.a
+modes(d::LogUniform)  = partype(d)[]
+
+#### Evaluation
+function pdf(d::LogUniform, x::Real)
+    x, a, b = promote(x, params(d)...)
```

Comment:
These promotions should not be needed, what is the motivation for them?

Reply:
So that `Float32` is preserved. E.g. `pdf(LogUniform(1f0,2f0), 3f0)::Float32`

Reply:
Yes, this is seems desirable. Can you add a comment that explains it? And use e.g.
```suggestion
    _x, a, b = promote(x, params(d)...)
```
(or some other variable name) to ensure that there are no type stability issues (sometimes the compiler chokes if assign a new value of a different type to the same variable name).